manage_breast_screening/notifications/management/commands/create_appointments.py

```diff
@@ -0,0 +1,107 @@
+import io
+import os
+from datetime import datetime
+from functools import cached_property
+from zoneinfo import ZoneInfo
+
+import pandas
+from azure.storage.blob import BlobServiceClient, ContainerClient
+from django.core.management.base import BaseCommand, CommandError
+
+from manage_breast_screening.notifications.models import Appointment, Clinic
+
+
+class Command(BaseCommand):
+    """
+    Django Admin command which reads NBSS appointment data from Azure blob storage
+    and saves data as Appointment and Clinic records in the database.
+
+    Requires the env vars `BLOB_STORAGE_CONNECTION_STRING` and `BLOB_CONTAINER_NAME`
+    to connect to Azure blob storage.
+    """
+
+    def handle(self, *args, **options):
+        try:
+            today_dirname = datetime.today().strftime("%Y-%m-%d")
+
+            for blob in self.container_client().list_blobs(
+                name_starts_with=today_dirname
+            ):
+                blob_client = self.container_client().get_blob_client(blob)
+                blob_content = blob_client.download_blob(
+                    max_concurrency=1, encoding="ASCII"
+                ).readall()
+
+                data_frame = self.raw_data_to_data_frame(blob_content)
+
+                for idx, row in data_frame.iterrows():
+                    clinic, clinic_created = self.find_or_create_clinic(row)
+                    if clinic_created:
+                        self.stdout.write(f"{clinic} created")
+
+                    appt, appt_created = self.find_or_create_appointment(row, clinic)
+                    if appt_created:
+                        self.stdout.write(f"{appt} created")
+
+                self.stdout.write(f"Processed {data_frame.size} rows from {blob.name}")
+        except Exception as e:
+            raise CommandError(e)
+
+    def raw_data_to_data_frame(self, raw_data: str) -> pandas.DataFrame:
+        return pandas.read_table(
+            io.StringIO(raw_data),
+            dtype="str",
+            encoding="ASCII",
+            engine="python",
+            header=1,
+            sep="|",
+            skipfooter=1,
+        )
+
+    def find_or_create_clinic(self, row: dict) -> Clinic:
```

Comment:
minor/nitpick, I got some type errors here, as `get_or_create` returns a tuple of [Clinic, bool] same for `find_or_create_appointment` below

Reply:
https://github.com/NHSDigital/dtos-manage-breast-screening/pull/179/commits/c83b7364630d7970e4eb6997c01c3b7074cf41f4 to address method return typehints